sample.config.yaml

```diff
@@ -20,51 +56,117 @@ provisioning:
   whitelist:
     # Allow a specific user
     #- "@user:server\\.com"
+
     # Allow users on a specific homeserver
-    - "@.*:yourserver\\.com"
+    - "@.*:server\\.com"
+
     # Allow anyone
     #- ".*"
+
   # Regex of Matrix IDs forbidden from using the puppet bridge
   #blacklist:
     # Disallow a specific user
     #- "@user:server\\.com"
+
     # Disallow users on a specific homeserver
-    #- "@.*:yourserver\\.com"
+    #- "@.*:server\\.com"
+
+relay:
+  # Regex of Matrix IDs who are allowed to use the bridge in relay mode.
+  # Relay mode is when a single Discord bot account relays messages of
+  # multiple Matrix users
+  #
+  # Same format as in provisioning
+  whitelist:
+    - "@.*:yourserver\\.com"
+
+  #blacklist:
+    #- "@user:yourserver\\.com"
+
+selfService:
+  # Regex of Matrix IDs who are allowed to use bridge self-servicing (plumbed rooms)
+  #
+  # Same format as in provisioning
+  whitelist:
+    - "@.*:server\\.com"
+
+  #blacklist:
+    #- "@user:server\\.com"
+
+# Map of homeserver URLs to their C-S API endpoint
+#
+# Useful for double-puppeting if .well-known is unavailable for some reason
+#homeserverUrlMap:
+  #yourserver.com: http://localhost:1234
+
+# Override the protocols set default name patterns
+#
+# Which variables are available depends on protocol implementation
```

Comment:
What are these exactly? Overriding the default for user, room and group displaynames? What's the difference between `user` and `userOverride`?

Since we know the protocol implementation here, what's available for Discord?